src/network/mcpe/InventoryManager.php

```diff
@@ -210,12 +233,17 @@ public function onClientRemoveWindow(int $id) : void{
 	}
 
 	public function syncSlot(Inventory $inventory, int $slot) : void{
-		$windowId = $this->getWindowId($inventory);
-		if($windowId !== null){
-			$currentItem = $inventory->getItem($slot);
+		$currentItem = $inventory->getItem($slot);
+		$itemStackWrapper = ItemStackWrapper::legacy(TypeConverter::getInstance()->coreItemStackToNet($currentItem));
+		if(($slotOffset = self::getSlotOffset($inventory)) !== null){
```

Comment:
This should check that the windowID is actually UI, in case a plugin mapped it to something else.